sys/net/gnrc/network_layer/ipv6/static_addr/gnrc_ipv6_static_addr.c

```diff
@@ -113,25 +113,30 @@ static void _config_downstream(gnrc_netif_t *downstream)
 
 void auto_init_gnrc_ipv6_static_addr(void)
 {
-    gnrc_netif_t *netif = NULL;
     gnrc_netif_t *upstream = NULL;
     gnrc_netif_t *downstream = NULL;
 
-    if (IS_ACTIVE(CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM)) {
-        upstream = gnrc_netif_get_by_pid(CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM);
-    }
+    if (gnrc_netif_highlander() || gnrc_netif_numof() == 1) {
+        upstream = gnrc_netif_iter(NULL);
```

Comment:
Would it make sense to at least issue a warning that this is might be `!= CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM` (i.e. effectively ignoring a potentially nonsensical configuration)? Or even have an assertion here?
```suggestion
        upstream = gnrc_netif_iter(NULL);
        if (IS_ACTIVE(CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM) &&
            (CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM != upstream->pid) {
            LOG_WARNING("Only netif %d differs from selected upstream netif %d\n", netif->pid,
                        CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM);
        }
```

Or

```suggestion
        upstream = gnrc_netif_iter(NULL);
        assert(!IS_ACTIVE(CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM) ||
               CONFIG_GNRC_IPV6_STATIC_ADDR_UPSTREAM == upstream);
```

Reply:
Well if you insist…
I really think this makes the code more complicated than it needs to be

Reply:
Does it? In any case it provides a user that might have misconfigured their system with valuable information why things don't work the way they expect it to be IMHO. The less complicated solution is the assert if you ask me.

Reply:
Thanks!